From 1b1ee1e18f5bb4d91e3839405a8557a64a3b2581 Mon Sep 17 00:00:00 2001
From: Mattias Ellert <mattias.ellert@physics.uu.se>
Date: Sat, 20 Jan 2024 08:47:28 +0100
Subject: [PATCH] erasure_code: fix wrong return type

erasure_code/ppc64le/gf_vect_mul_vsx.c: In function '_gf_vect_mul_base':
erasure_code/ppc64le/gf_vect_mul_vsx.c:14:16: error: 'return' with a value, in function returning void [-Wreturn-mismatch]
   14 |         return 0;
      |                ^
erasure_code/ppc64le/gf_vect_mul_vsx.c:6:13: note: declared here
    6 | static void _gf_vect_mul_base(int len, unsigned char *a, unsigned char *src,
      |             ^~~~~~~~~~~~~~~~~

Signed-off-by: Mattias Ellert <mattias.ellert@physics.uu.se>
---
 erasure_code/ppc64le/gf_vect_mul_vsx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/erasure_code/ppc64le/gf_vect_mul_vsx.c b/erasure_code/ppc64le/gf_vect_mul_vsx.c
index 0f2f2f46..718d96ce 100644
--- a/erasure_code/ppc64le/gf_vect_mul_vsx.c
+++ b/erasure_code/ppc64le/gf_vect_mul_vsx.c
@@ -11,7 +11,7 @@ static void _gf_vect_mul_base(int len, unsigned char *a, unsigned char *src,
 
 	while (len-- > 0)
 		*dest++ = gf_mul(c, *src++);
-	return 0;
+	return;
 }
 
 void gf_vect_mul_vsx(int len, unsigned char *gftbl, unsigned char *src, unsigned char *dest)
